### DISCUSSION TITLE
Add Support for Luajit

### DISCUSSION THREAD

**Discussion by GPeaky** [citizenfx/fivem#2363] [Engineering Discussion]
Hi, i have seen that Fivem have support for lua 5.3 and 5.4, the majority of the scripts of fivem are doned with lua. And some servers lacks performance, so why don't implement Luajit that is much faster than lua5.4?

**This is just an example code:**

```lua
local t = {}

for i = 1, 1000000 do
    t[i] = i
end

local sum = 0
for i = 1, 1000000 do
    sum = sum + t[i]
end

print(sum)
```

**Lua54 Result:**
```bash
500000500000
Elapsed: 15.4517ms
```

**LuaJit Result:**
```bash
500000500000
Elapsed: 3.4772ms
```

This test don't show the real case difference but its a showcase to see the performance difference between the two options. And i think if this is implement most of the servers and clients will notice it.

**Comment by andreiciobanica** [comment]
Bump?

**Comment by n3xuuu** [comment]
would love to see this become a reality =)

**Comment by rol3x1337** [comment]
i don't think that will ever happen, but bump 🙏

**Comment by ExtReMLapin** [comment]
LuaJIT nerd here, you misunderstand how things works.

Here your code is jit compiled. Even if the luajit interpreter is 4 times faster here you're using the jit compiler which will not jit compile your code because of all of the LUA C API call inside your loops.

LuaJIT supports stitching which kinds helps but still, you would need to use FFI to 100% use luajit performance
